File: payment-service/src/main/java/com/study/payment/application/dto/PaymentResponse.java
package com.study.payment.application.dto;

import com.study.payment.domain.entity.Payment;
import com.study.payment.domain.entity.PaymentStatus;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Builder
public class PaymentResponse {
    private Long id;
    private Long bookingId;
    private Long userId;
    private BigDecimal amount;
    private PaymentStatus status;
    private LocalDateTime paidAt;
    private LocalDateTime createdAt;

    public static PaymentResponse from(Payment payment) {
        return PaymentResponse.builder()
                .id(payment.getId())
                .bookingId(payment.getBookingId())
                .userId(payment.getUserId())
                .amount(payment.getAmount())
                .status(payment.getStatus())
                .paidAt(payment.getPaidAt())
                .createdAt(payment.getCreatedAt())
                .build();
    }
}

File: payment-service/src/main/kotlin/com/study/payment/application/dto/PaymentResponse.kt
package com.study.payment.application.dto

import com.study.payment.domain.entity.Payment
import com.study.payment.domain.entity.PaymentStatus
import java.math.BigDecimal
import java.time.LocalDateTime

data class PaymentResponse(
    val id: Long, val bookingId: Long, val userId: Long,
    val amount: BigDecimal, val status: PaymentStatus,
    val paidAt: LocalDateTime?, val createdAt: LocalDateTime
) {
    companion object {
        fun from(payment: Payment) = PaymentResponse(
            id = payment.id, bookingId = payment.bookingId, userId = payment.userId,
            amount = payment.amount, status = payment.status,
            paidAt = payment.paidAt, createdAt = payment.createdAt
        )
    }
}
